Source/Core/Gen3/Searchers/WildSearcher3.cpp

```diff
@@ -97,8 +97,28 @@ std::vector<WildState> WildSearcher3::search(u8 hp, u8 atk, u8 def, u8 spa, u8 s
         return states;
     }
 
-    bool rseSafariRockSmash = encounterArea.rseSafariZone() && rse && encounter == Encounter::RockSmash; // RockSmash encounters have different rng calls inside RSE Safari Zone,
-                                                                                                         // so we set a flag to check if we're searching these kind of spreads
+    // RockSmash/Surfing/Fishing encounters have different rng calls inside RSE Safari Zone,
+    // so we set a flag to check if we're searching these kind of spreads
+    bool rseSafari = encounterArea.rseSafariZone() && rse;
+    bool rseSafariRockSmash = false;
+    bool rseSafariSurfingFishing = false;
+    if (rseSafari)
+    {
+        switch (encounter)
+        {
+        case Encounter::RockSmash:
+            rseSafariRockSmash = true;
+            break;
+        case Encounter::Surfing:
+        case Encounter::OldRod:
+        case Encounter::GoodRod:
+        case Encounter::SuperRod:
+            rseSafariSurfingFishing = true;
+            break;
+        default:
+            break;
+        }
+    }
```

Comment:
```
bool rseSafari = encounterArea.rseSafariZone() & rse;
switch (encounter)
{
case Encounter::RockSmash:
case Encounter::Surfing:
case Encounter::OldRod:
case Encounter::GoodRod:
case Encounter::SuperRod:
    rseSafari &= true;
    break;
default:
    break;
}
```

Also make sure to have a blank line after the switch statement

Reply:
I completely removed it cause i noticed it is not needed. I made many tests and everything is perfectly working now!

Reply:
How, you added this code with the previous PR and now its not needed?

Reply:
> How, you added this code with the previous PR and now its not needed?

Safari wild searcher in RSE has missing calls, it is broken in main branch. I noticed it only after adding this, now everything is fixed

Reply:
How would it not break non safari case?

Reply:
Cause there is the rseSafari check

Reply:
What about the grass case in safari. Pretty sure that would be incorrerct